src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/CaseSearchRequestFactory.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.ccd.domain.service.search.elasticsearch;
+
+import static java.lang.String.format;
+
+import uk.gov.hmcts.ccd.ApplicationParams;
+
+public abstract class CaseSearchRequestFactory<T> {
```

Comment:
are we operating on the overall search request in this class or only on the query? In the latter case, maybe better to rename this to `CaseSearchQueryFactory`
and its method to `createSearchQuery`

Reply:
also, I'm thinking that the names factory and create are a bit misleading. Because we are not creating the query. The query comes from the client. 
We are transforming the query. So perhaps we should rename these to `CaseSearchQueryTransformer` and `transformSearchQuery`?

Reply:
any reason why this is an abstract class with createSearchRequest not implemented directly here?

Reply:
how come this is  a generic class?

Reply:
the above seems a bit an uneeded complication to me. I would remove them and go for a simple concrete non generic class. If in the future the need arise, we can make it generic and create a hierarchy.
Unless I'm missing something

Reply:
Was just trying to enforce the security method is called. Although I agree, its unncessary complication. I have removed it now and simplified the code.